service/interpreter/InternalChannel.go

```diff
@@ -0,0 +1,57 @@
+package interpreter
+
+import "github.com/indeedeng/iwf/gen/iwfidl"
+
+type InternalChannel struct {
+	// key is channel name
+	receivedData map[string][]*iwfidl.EncodedObject
+}
+
+func NewInterStateChannel() *InternalChannel {
+	return &InternalChannel{
+		receivedData: map[string][]*iwfidl.EncodedObject{},
+	}
+}
+
+func RebuildInterStateChannel(refill map[string][]*iwfidl.EncodedObject) *InternalChannel {
+	return &InternalChannel{
+		receivedData: refill,
+	}
+}
+
+func (i *InternalChannel) DumpReceived() map[string][]*iwfidl.EncodedObject {
```

Comment:
Should this be GetReceived? Or just `Retrieve` and then `Retrieve(channelName string)` -> `RetrieveByChannel(channelName string)`?

Reply:
👍 
I think `GetAllReceived` maybe more clear and consistent with other `GetAll...` methods for continueAsNewer to call.